native/sqlparser_ex/src/lib.rs

```diff
@@ -19,40 +19,85 @@ struct Response {
     message: String,
 }
 
+impl Response {
+    fn ok(message: String) -> Self {
+        Self {
+            status: atoms::ok(),
+            message,
+        }
+    }
+
+    fn error(message: &str) -> Self {
+        Self {
+            status: atoms::error(),
+            message: message.to_string(),
+        }
+    }
+}
+
 #[rustler::nif]
 fn parse(dialect_str: &str, query: &str) -> NifResult<Response> {
     let result = match dialect_str {
         "bigquery" => Parser::parse_sql(&BigQueryDialect {}, query),
+        "clickhouse" => Parser::parse_sql(&ClickHouseDialect {}, query),
         "postgres" => Parser::parse_sql(&PostgreSqlDialect {}, query),
-        _ => Err(ParserError(
-            "Parser for this dialect is not supported.".to_string(),
-        )),
+        _ => return Ok(Response::error("Parser for this dialect is not supported.")),
     };
+
     match result {
-        Ok(v) => Ok(Response {
-            status: atoms::ok(),
-            message: serde_json::to_string(&v).unwrap(),
-        }),
-        Err(v) => Ok(Response {
-            status: atoms::error(),
-            message: v.to_string(),
-        }),
+        Ok(ast) => match serde_json::to_string(&ast) {
+            Ok(json) => Ok(Response::ok(json)),
+            Err(e) => Ok(Response::error(&format!("JSON serialization error: {}", e))),
+        },
+        Err(e) => Ok(Response::error(&e.to_string())),
+    }
+}
+
+fn ensure_required_fields(value: &mut serde_json::Value) {
+    use serde_json::Value;
+
+    match value {
+        Value::Object(obj) => {
+            for key in ["Query", "Select", "Function"] {
+                if let Some(Value::Object(nested)) = obj.get_mut(key) {
+                    nested.entry("order_by").or_insert_with(|| Value::Array(vec![]));
+                }
+            }
+
+            for (_, v) in obj.iter_mut() {
+                ensure_required_fields(v);
+            }
+        },
+        Value::Array(arr) => {
+            for item in arr.iter_mut() {
+                ensure_required_fields(item);
+            }
+        },
+        _ => {}
     }
 }
 
 #[rustler::nif]
 fn to_string(json: &str) -> NifResult<Response> {
-    let nodes: Vec<sqlparser::ast::Statement> = serde_json::from_str(json).unwrap();
+    let mut value: serde_json::Value = match serde_json::from_str(json) {
+        Ok(val) => val,
+        Err(e) => return Ok(Response::error(&format!("JSON parsing error: {}", e))),
+    };
 
-    let mut parts = vec![];
-    for node in nodes {
-        parts.push(format!("{}", node))
-    }
+    ensure_required_fields(&mut value);
```

Comment:
i think we can revert the this validation logic, seems unnecessary and makes the function more restrictive.

Reply:
ok